src/py/mat3ra/made/tools/build/slab/builders.py

```diff
@@ -29,79 +26,75 @@ class PymatgenSlabGeneratorParameters(PymatgenSlabGeneratorParametersSchema):
-            symmetrize=build_parameters.symmetrize
-        )
-        self.__configuration = configuration
-
-        return [self.__conditionally_convert_slab_to_orthogonal_pymatgen(slab) for slab in raw_slabs]
-
-    def _select(
-        self, items: List[_GeneratedItemType], selector_parameters: _SelectorParametersType
-    ) -> List[_GeneratedItemType]:
-        termination = selector_parameters.termination
-        return [slab for slab in items if self.__create_termination_from_slab_pymatgen(slab) == termination]
-
-    def _post_process(self, items: List[_GeneratedItemType], post_process_parameters=None) -> List[Material]:
-        materials = super()._post_process(items, post_process_parameters)
-        materials = [create_supercell(material, self.__configuration.xy_supercell_matrix) for material in materials]
-        build_parameters = self.build_parameters or SlabBuilderParameters()
-
-        # Adding total vacuum to be exactly as specified in configuration, including already added vacuum
-        added_vacuum = (
-            build_parameters.min_vacuum_size * self.__configuration.bulk.lattice.c
-            if build_parameters.in_unit_planes
-            else build_parameters.min_vacuum_size
-        )
-        vacuum_to_add = self.__configuration.vacuum - added_vacuum
-
-        materials_with_vacuum = [add_vacuum(material, vacuum_to_add) for material in materials]
-        for idx, material in enumerate(materials_with_vacuum):
-            if "build" not in material.metadata:
-                material.metadata["build"] = {}
-            material.metadata["build"]["termination"] = label_pymatgen_slab_termination(items[idx])
 
-        return materials_with_vacuum
+    def get_material(self, configuration: _ConfigurationType) -> Material:  # type: ignore
+        self._configuration = configuration
+        return super().get_material(configuration)
 
     def get_terminations(self, configuration: _ConfigurationType) -> List[Termination]:
-        return [
-            self.__create_termination_from_slab_pymatgen(slab)
-            for slab in self._generate_or_get_from_cache(configuration)
-        ]
-
-    def __conditionally_convert_slab_to_orthogonal_pymatgen(self, slab: PymatgenSlab) -> PymatgenSlab:
-        return slab.get_orthogonal_c_slab() if self.__configuration.use_orthogonal_z else slab
+        """Get available terminations for the given configuration."""
+        atomic_layers: AtomicLayersUniqueRepeated = configuration.stack_components[0]
+        return atomic_layers.get_terminations()
+
+    def _generate(self, configuration: _ConfigurationType) -> List[Material]:  # type: ignore
+        atomic_layers: AtomicLayersUniqueRepeated = configuration.stack_components[0]
+        vacuum: VacuumConfiguration = configuration.stack_components[1]
+        params = self.build_parameters or SlabBuilderParameters()
+
+        if params.use_orthogonal_c:
+            pymatgen_slabs = atomic_layers._generate_pymatgen_slabs(
```

Comment:
This logic about `use_orthogonal_c` should to `atomic_layers` here we should just call `atomic_layers.get_slabs(use_orthogonal_c=True)`

Reply:
Make it so we can pass *args, **kwargs to extra_parameters